### PR TITLE
Refactoring.

### PR DESCRIPTION
**Pull request by PradeepKadubandi**
More Refactoring.

### CODE REVIEW

File: src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs [hunk 1]
```diff
@@ -62,77 +64,72 @@ public virtual IEnumerable<string> TemplateFolders
...
-            ModelType dataContext = ValidationUtil.ValidateType(viewGeneratorModel.DataContextClass, "dataContext", _modelTypesLocator, throwWhenNotFound: false);
-
-            // Validation successful
-            Contract.Assert(model != null, "Validation succeded but model type not set");
-
-            var appbasePath = _applicationEnvironment.ApplicationBasePath;
-            var outputFolder = String.IsNullOrEmpty(viewGeneratorModel.RelativeFolderPath)
-                ? Path.Combine(appbasePath)
-                : Path.Combine(appbasePath, viewGeneratorModel.RelativeFolderPath);
+            var outputPath = ValidateAndGetOutputPath(viewGeneratorModel, outputFileName: viewGeneratorModel.ViewName + Constants.ViewExtension);
```
Comment (abpiskunov) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs]: "Views" folder in the path is not added (in base class also)?

File: src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs [hunk 2]
```diff
@@ -62,77 +64,72 @@ public virtual IEnumerable<string> TemplateFolders
...
+                ViewDataTypeName = modelTypeAndContextModel.ModelType.FullName,
+                ViewDataTypeShortName = modelTypeAndContextModel.ModelType.Name,
                 ViewName = viewGeneratorModel.ViewName,
                 LayoutPageFile = viewGeneratorModel.LayoutPage,
                 IsLayoutPageSelected = isLayoutSelected,
                 IsPartialView = viewGeneratorModel.PartialView,
                 ReferenceScriptLibraries = viewGeneratorModel.ReferenceScriptLibraries,
-                ModelMetadata = modelMetadata,
+                ModelMetadata = modelTypeAndContextModel.ModelMetadata,
                 JQueryVersion = "1.10.2" //Todo
```
Comment (abpiskunov) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs]: how is this tracked?

**Comment by abpiskunov** [comment]
:watch:

File: src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs (hunk 1 above)
Comment (PradeepKadubandi) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs]: The behavior in VS is that view is created in the folder where the context menu is invoked, so I removed any assumptions about conventions in command line code and only using RelativePath if provided as the single options for modifying the output path. VS will pass the right value for this parameter.

File: src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs (hunk 2 above)
Comment (PradeepKadubandi) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs]: You mean the JQueryVersion? Yeah, the scripts issues that I logged should be used to fix this...As of now, I think the parameter is not even used in templates ... we need to decide what scripts do we want to reference and what version and accordingly fix this...

**Pull request merged**